src/v2i-hub/TimPlugin/src/TimPlugin.cpp

```diff
@@ -0,0 +1,438 @@
+
+	ostringstream lastTimTime_;
+        lastTimTime_ << _stopTimDate << " " << _endTime;
+        auto lastTime = lastTimTime_.str();
+
+        istringstream stopTimDate(lastTime);
+
+	struct tm date_stop;
+	stopTimDate >> get_time( &date_stop, "%m-%d-%Y %H:%M:%S" );
+	time_t secondsStopDate = mktime( & date_stop );
+
+        auto t = time(nullptr);
+        auto tm = *localtime(&t);
+
+        ostringstream oss1;
+        oss1 << put_time(&tm, "%m-%d-%Y");
+        auto _currentTimDate = oss1.str();
+
+	istringstream currentTimDate(_currentTimDate);
+
+	struct tm date_current;
+	currentTimDate >> get_time( &date_current, "%m-%d-%Y" );
+	time_t secondsCurrentDate = mktime( & date_current );
+
+        ostringstream oss2;
+        oss2 << put_time(&tm, "%H:%M:%S");
+        auto _currentTimTime = oss2.str();
+
+	ostringstream currentTimTime_;
+        currentTimTime_ << _currentTimDate << " " << _currentTimTime;
+        auto currentTime = currentTimTime_.str();
+
+        ostringstream startTimTime_;
+        startTimTime_ << _currentTimDate << " " << _startTimTime;
+        auto StartTime = startTimTime_.str();
+
+        ostringstream stopTimTime_;
+        stopTimTime_ << _currentTimDate << " " << _stopTimTime;
+        auto StopTime = stopTimTime_.str();
+
+        istringstream currentTimTime(currentTime);
+        istringstream StartTimTime(StartTime);
+        istringstream StopTimTime(StopTime);
+
+	struct tm time_current;
+	currentTimTime >> get_time( &time_current, "%m-%d-%Y %H:%M:%S" );
+	time_t secondsCurrentTime = mktime( & time_current );
+
+	struct tm time_start;
+	StartTimTime >> get_time( &time_start, "%m-%d-%Y %H:%M:%S" );
+	time_t secondsStartTime = mktime( & time_start );
+
+	struct tm time_stop;
+	StopTimTime >> get_time( &time_stop, "%m-%d-%Y %H:%M:%S" );
+	time_t secondsStopTime = mktime( & time_stop );
+
+	if ((secondsStartDate <= secondsCurrentDate) && (secondsCurrentDate <= secondsStopDate) && (secondsStartTime <= secondsCurrentTime) && (secondsCurrentTime <= secondsStopTime)) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+bool TimPlugin::LoadTim(TravelerInformation *tim, const char *mapFile)
+{
+	memset(tim, 0, sizeof(TravelerInformation));
+
+	// J2735 packet header.
+
+	//tim->msgID = DSRCmsgID_travelerInformation;
+
+	DsrcBuilder::SetPacketId(tim);
+
+	// Data Frame (1 of 1).
+
+	XmlCurveParser curveParser;
+
+	std::cout << "Loading curve file: " << mapFile << std::endl;
+
+	// Read the curve file, which creates and populates the data frame of the TIM.
+	if (!curveParser.ReadCurveFile(mapFile, tim))
+		return false;
+
+	// Verify that a single data frame was added by the parser.
+	//if (tim->dataFrameCount == NULL || *tim->dataFrameCount != 1)
+	//	return false;
+
+	std::cout << "TIM was created." << std::endl;
+
+	_speedLimit = curveParser.SpeedLimit;
+
+	PluginUtil::SetStatus<unsigned int>(_plugin, "Speed Limit", _speedLimit);
+
+	return true;
+}
+
+void TimPlugin::TestFindRegion()
```

Comment:
Is this needed?

Reply:
I am leaving this in, I haven't been able to test it's impact. I will include it in the cleanup for the web service story.